// @novu/shared/dist/cjs/utils/buildWorkflowPreferences.spec.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
const vitest_1 = require("vitest");
const buildWorkflowPreferences_1 = require("./buildWorkflowPreferences");
const WORKFLOW_CHANNEL_PREFERENCE_DEFAULT_VALUE = true;
const WORKFLOW_CHANNEL_PREFERENCE_DEFAULT_READ_ONLY = false;
const DEFAULT_WORKFLOW_PREFERENCE = {
    enabled: WORKFLOW_CHANNEL_PREFERENCE_DEFAULT_VALUE,
    readOnly: WORKFLOW_CHANNEL_PREFERENCE_DEFAULT_READ_ONLY,
};
const DEFAULT_CHANNEL_PREFERENCE = {
    enabled: WORKFLOW_CHANNEL_PREFERENCE_DEFAULT_VALUE,
};
const testDefaultPreferences = {
    all: DEFAULT_WORKFLOW_PREFERENCE,
    channels: {
        in_app: DEFAULT_CHANNEL_PREFERENCE,
        sms: DEFAULT_CHANNEL_PREFERENCE,
        email: DEFAULT_CHANNEL_PREFERENCE,
        push: DEFAULT_CHANNEL_PREFERENCE,
        chat: DEFAULT_CHANNEL_PREFERENCE,
    },
};
(0, vitest_1.describe)('buildWorkflowPreferences', () => {
    (0, vitest_1.it)('should return the defaults if input is undefined', () => {
        const result = (0, buildWorkflowPreferences_1.buildWorkflowPreferences)(undefined, testDefaultPreferences);
        (0, vitest_1.expect)(result).toEqual(testDefaultPreferences);
    });
    (0, vitest_1.it)('should return the input object if a complete preferences object is supplied', () => {
        const testWorkflowPreference = {
            enabled: false,
            readOnly: true,
        };
        const testChannelPreference = {
            enabled: false,
        };
        const testPreferences = {
            all: testWorkflowPreference,
            channels: {
                in_app: testChannelPreference,
                sms: testChannelPreference,
                email: testChannelPreference,
                push: testChannelPreference,
                chat: testChannelPreference,
            },
        };
        const result = (0, buildWorkflowPreferences_1.buildWorkflowPreferences)(testPreferences, testDefaultPreferences);
        (0, vitest_1.expect)(result).toEqual(testPreferences);
    });
    (0, vitest_1.describe)('should populate the remainder of the object with default values', () => {
        (0, vitest_1.it)('using just a full, single channel', () => {
            const testPreferences = {
                channels: { in_app: { enabled: false } },
            };
            const result = (0, buildWorkflowPreferences_1.buildWorkflowPreferences)(testPreferences, testDefaultPreferences);
            (0, vitest_1.expect)(result).toEqual(Object.assign(Object.assign({}, testDefaultPreferences), { channels: Object.assign(Object.assign({}, testDefaultPreferences.channels), { in_app: { enabled: false } }) }));
        });
        (0, vitest_1.it)('using a combination of channels and workflow-level preferences', () => {
            var _a, _b, _c;
            const testPreferences = {
                all: { enabled: true, readOnly: true },
                channels: {
                    in_app: { enabled: false },
                    chat: { enabled: false },
                },
            };
            const result = (0, buildWorkflowPreferences_1.buildWorkflowPreferences)(testPreferences, testDefaultPreferences);
            (0, vitest_1.expect)(result).toEqual({
                all: testPreferences.all,
                channels: {
                    in_app: {
                        enabled: false,
                    },
                    chat: {
                        enabled: false,
                    },
                    sms: {
                        enabled: (_a = testPreferences.all) === null || _a === void 0 ? void 0 : _a.enabled,
                    },
                    email: {
                        enabled: (_b = testPreferences.all) === null || _b === void 0 ? void 0 : _b.enabled,
                    },
                    push: {
                        enabled: (_c = testPreferences.all) === null || _c === void 0 ? void 0 : _c.enabled,
                    },
                },
            });
        });
    });
    (0, vitest_1.it)('should use the `workflow`-level preferences to define defaults for all channel-level preferences', () => {
        const expectedDefaultValue = false;
        const testPreferences = {
            all: { enabled: expectedDefaultValue },
        };
        const result = (0, buildWorkflowPreferences_1.buildWorkflowPreferences)(testPreferences, testDefaultPreferences);
        const expectedResult = {
            all: {
                enabled: expectedDefaultValue,
                readOnly: WORKFLOW_CHANNEL_PREFERENCE_DEFAULT_READ_ONLY,
            },
            channels: {
                in_app: {
                    enabled: expectedDefaultValue,
                },
                sms: {
                    enabled: expectedDefaultValue,
                },
                email: {
                    enabled: expectedDefaultValue,
                },
                push: {
                    enabled: expectedDefaultValue,
                },
                chat: {
                    enabled: expectedDefaultValue,
                },
            },
        };
        (0, vitest_1.expect)(result).toEqual(expectedResult);
    });
});
