// @novu/node/build/module/lib/workflow-override/workflow-override.js
import { WithHttp } from '../novu.interface';
export class WorkflowOverrides extends WithHttp {
    async updateOneById(overrideId, data) {
        return await this.http.put(`/workflow-overrides/${overrideId}`, {
            ...data,
        });
    }
    async create(data) {
        return await this.http.post(`/workflow-overrides`, {
            ...data,
        });
    }
    async getOneById(overrideId) {
        return await this.http.get(`/workflow-overrides/${overrideId}`);
    }
    async list(page = 0, limit = 10) {
        return await this.http.get(`/workflow-overrides`, {
            params: { page, limit },
        });
    }
    async getOneByTenantIdandWorkflowId(workflowId, tenantId) {
        return await this.http.get(`/workflow-overrides/workflows/${workflowId}/tenants/${tenantId}`);
    }
    async updateOneByTenantIdandWorkflowId(workflowId, tenantId, data) {
        return await this.http.put(`/workflow-overrides/workflows/${workflowId}/tenants/${tenantId}`, {
            ...data,
        });
    }
    async delete(overrideId) {
        return await this.http.delete(`/workflow-overrides/${overrideId}`);
    }
}
